app/helpers/application_helper.rb

```diff
@@ -15,4 +15,54 @@ def like_or_dislike_btn(post)
       link_to('Like!', post_likes_path(post_id: post.id), method: :post)
     end
   end
+
+  def friendship(ids)
+    friend = User.find_by_id(ids)
+    friendship = Friendship.find_by(user: current_user, friend: friend)
+    inverse_friendship = Friendship.find_by(user: friend, friend: current_user)
+    if friendship.nil? && inverse_friendship.nil?
+      link_to('Send Friend request', new_user_friendship_path(ids))
+    else destroy_friend_request(friend.id)
+    end
+  end
+
+  def destroy_friendship(ids)
+    friend = User.find_by_id(ids)
```

Comment:
You seem to have application logic inside of your helpers. Helper methods ought to be view specific; the fact that you have to import the helpers in Application Controller for you to have access to those methods in controllers while you do not even use it in your views reiterates this. The views on the other hand, automatically have access to helper methods. 

So any `User.find.by` or `Friendship.find_by` etc. should not be in helpers
So, what you can do is add all methods which are not views specific to your application controller directly. What you can also do if your method list gets long, you may use concerns or a lib folder to separate methods by function, etc.

Basically helpers = views, controller = controller. Why write methods in helpers only to have to include it in controller after. Helpers are automatically accessible in views. there is a reason helpers are outside the context of the controller folder. You can, however, use concerns in your controllers.

Read [this article](http://codefol.io/posts/Where-Do-I-Put-My-Code/) and [this article](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) for a clearer explanation. 🙂 

**Not sure this was a requirement?** Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from [pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/).

Reply:
Great suggestions, thank you I couldn't see it that way Right now it's all solved and it looks nicer than before